### ISSUE TITLE
Windows build is broken

### ISSUE THREAD

**Issue by amolenaar** [gaphor/gaphor#1122]
### Describe the bug

The windows build works for Gaphor 2.6.5, but on master it's broken.

### To Reproduce
1. Install Gaphor from the windows installer
2. Try to start it.
3. Check the logbook for a  startup issue (SideBySide)

### OS
- [ ] Linux (Please put in notes the specific distro)
- [ ] macOS
- [X] Windows

### Version
Version of Gaphor: master 

### Additional information

* FAIL - https://github.com/gaphor/gaphor/actions/runs/1446986173 (1feb41f)
* OK - https://github.com/gaphor/gaphor/actions/runs/1442111507 (bee8453c)
* OK - https://github.com/gaphor/gaphor/actions/runs/1441080609 (7edd6ba3)
* OK - https://github.com/gaphor/gaphor/actions/runs/1432108555 (f19332e)
* OK - https://github.com/gaphor/gaphor/actions/runs/1420303387 (b61bef74)

The first failing build says it's an update of Sphinx 🤷

**Comment by danyeaw** [comment]
Weird. I am able to freeze it locally with PyInstaller, and it works fine. When I try to launch the gaphor-exe.exe I get the following error:

![image](https://user-images.githubusercontent.com/10014976/141699917-f2a0f0ea-2463-4c8b-bc14-374ee1128327.png)

It is almost like the NSIS installer is broken. Running `sfc /scannow` didn't find any integrity violations.

**Comment by amolenaar** [comment]
That's the same message I got. Only I got it in Dutch, so I figured it's probably not worth posting :wink:

**Comment by danyeaw** [comment]
It looks like this issue was caused by a [regression](https://bugs.python.org) in Python 3.9.8 which was released on 2021-11-05. So I don't think this was related to any change we made, but more like CI started to use the newer version of Python. This checks with the failed builds you found, the first one that wasn't working was when the change to Python 3.9.8 occurred.

PyInstaller implemented a workaround here: https://github.com/pyinstaller/pyinstaller/pull/6367. However, this will take time to result in a PyInstaller release.

Let me see if we can run a build now with Python 3.10 now that it has been out a bit longer, it doesn't have this regression problem.

**Comment by danyeaw** [comment]
Python 3.10 is seg faulting when running pytest in Linux: https://github.com/gaphor/gaphor/runs/4220137795

**Comment by danyeaw** [comment]
One more update, I confirmed that this bug is resolved when using Python 3.10 for the build in Windows. If we can resolve this pytest segfault in Linux then we would be all set.

**Comment by amolenaar** [comment]
> Python 3.10 is seg faulting when running pytest in Linux: https://github.com/gaphor/gaphor/runs/4220137795

I expect this is an issue on Azure DevOps/GitHub Actions side.

**Comment by danyeaw** [comment]
```
Starting program: /home/runner/work/gaphor/gaphor/.venv/bin/python -m pytest
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffecd36700 (LWP 5413)]
[New Thread 0x7fffebd35700 (LWP 5414)]
[New Thread 0x7fffead34700 (LWP 5415)]
[New Thread 0x7fffe9d33700 (LWP 5416)]
[New Thread 0x7fffdbfff700 (LWP 5417)]
[New Thread 0x7fffdaffe700 (LWP 5418)]
[New Thread 0x7fffd9ffd700 (LWP 5419)]
[New Thread 0x7fffc7fff700 (LWP 5420)]
[Detaching after fork from child process 5421]

Thread 1 "python" received signal SIGSEGV, Segmentation fault.
__GI___libc_free (mem=0x20) at malloc.c:3102
3102	malloc.c: No such file or directory.
```

This reminds me of an issue from a couple of years ago we had with Python 3.7: https://github.com/actions/virtual-environments/issues/159
Unfortunately, our solution was to move to Python 3.8 on CI 😄

**Comment by amolenaar** [comment]
I had a similar dejavu 😆

**Comment by amolenaar** [comment]
@danyeaw Is this the line that makes it all crash?

https://github.com/danyeaw/plugin-console-test/blob/master/plugin/console.py#L2

Or is it the combination of `gi` + `rlcompleter`?

We've had trouble with this library before. Although I think that was on Windows then.

**Comment by amolenaar** [comment]
It would be nice if we had an option to test the binaries for Window and macOS. 

E.g. start with a `self-test` option that will launch the application, do 1 or 2 simple checks and quit.

**Comment by danyeaw** [comment]
@amolenaar I agree, that's a great idea if we could automate that.

**Comment by danyeaw** [comment]
Closed by #1142. Thanks for reporting the issue @sz332!
